### PR TITLE
[VL] Add test for shuffle function

### PR DESCRIPTION
**Pull request by zhli1142015**
## What changes were proposed in this pull request?
Add test for shuffle function.

## How was this patch tested?

UT.

### CODE REVIEW

**Comment by zhli1142015** [comment]
cc @PHILO-HE and @rui-mo , thanks

**Review by rui-mo — APPROVED** [review]
Thanks.

File: gluten-core/src/main/scala/org/apache/gluten/backendsapi/SparkPlanExecApi.scala [hunk 1]
```diff
@@ -211,6 +211,13 @@ trait SparkPlanExecApi {
     GenericExpressionTransformer(substraitExprName, Seq(), original)
   }
 
+  def genShuffleTransformer(
+      substraitExprName: String,
+      child: ExpressionTransformer,
+      original: Shuffle): ExpressionTransformer = {
+    GenericExpressionTransformer(substraitExprName, Seq(child), original)
```
Comment (PHILO-HE) [inline gluten-core/src/main/scala/org/apache/gluten/backendsapi/SparkPlanExecApi.scala:218]: Why not pass seed? I guess CH backend also needs the seed. Then, we can remove genShuffleTransformer in API class.

cc @exmy, I note `shuffle` is mapped to `arrayshuffle` in CH. The doc says it supports seed. See https://clickhouse.com/docs/en/sql-reference/functions/array-functions#arrayshuffle

Comment (exmy) [reply gluten-core/src/main/scala/org/apache/gluten/backendsapi/SparkPlanExecApi.scala:218]: Look it's ok to pass seed for CH backend, I will check it later. @PHILO-HE

**Pull request merged**